### ISSUE TITLE
numpy.logical_or gives false error assignment-from-no-return

### ISSUE THREAD

**Issue by mhooreman** [PyCQA/pylint#2694]
### Steps to reproduce
Execute pylint on the following code:

```python
"""Demo of the assignment-from-no-return pylint false positive"""

import numpy as np


def demo():
    """Demo of the false positive, at line z = ..."""
    x = np.array([0, 1, 0, 1], dtype=np.bool)
    y = np.array([0, 0, 1, 1], dtype=np.bool)
    z = np.logical_or(x, y)
    print(np.array([x, y, z]))


if __name__ == "__main__":
    demo()
```

### Current behavior

```
assignment-from-no-return(E1111): test.py:10:4: demo: Assigning result of a function call, where the function has no return
```

### Expected behavior

No issue

### pylint --version output
```
pylint 2.2.2
astroid 2.1.0
Python 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34) 
[GCC 7.3.0]
```

**Comment by PCManticore** [comment]
Thanks for the report!

**Comment by arekfu** [comment]
For the record, this affects other functions too, such as `numpy.equal`.

**Comment by pxl-th** [comment]
`numpy.log` as well

**Comment by jordyjwilliams** [comment]
Any update as to the cause of this? or what versions this didn't occur on? could be a numpy issue?

**Comment by janosh** [comment]
Does anyone know if this needs fixing on `pylint`'s or `numpy`'s part?

**Comment by PCManticore** [comment]
This is on `pylint`'s side.

**Comment by jordyjwilliams** [comment]
@PCManticore any idea on when this will be fixed?

**Comment by raaaaaymond** [comment]
Can confirm same problem for a lot more Numpy functions, for example `deg2rad()`, `arctan()`, and more presumably. Moreover, array item assignment (such as `arr[i] = x`) suffers from `unsupported-assignment-operation`.

**Comment by jordyjwilliams** [comment]
+1 to all these issues

**Comment by PCManticore** [comment]
Should be fixed with the latest `master` of astroid.

**Comment by janosh** [comment]
@PCManticore That was quick! [The commit](https://github.com/PyCQA/astroid/commit/9d5925fe603c9b6dd98477d638dd397d4226029a#diff-02f0b547c2779d25cff89672135f20e3) says release date tba. Are we talking days or weeks?

**Comment by PCManticore** [comment]
Weeks most likely, I'll do a dev release in a bit, but the final one might be somewhere in February.

**Comment by IonyM** [comment]
Hi!

I'm facing the same issue in python2 within vscode.
tried updating both pylint and astroid but got already up-to-date, while there is a newer version available. Are this just foir Py3?
How can I solve the problem for Py2?

thanks a lot!

**Comment by Pierre-Sassoulas** [comment]
Sorry, the last minor versions supporting Python 2 are 1.9.x. We're currentely only supporting Python 3.6 and up (we're following what Python itself is doing).
